nixos/modules/services/web-apps/plausible.nix

```diff
@@ -0,0 +1,273 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.plausible;
+
+  # FIXME consider using LoadCredential as soon as it actually works.
+  envSecrets = ''
+    export ADMIN_USER_PWD="$(<${cfg.adminUser.passwordFile})"
```

Comment:
As pretty much 100% of shell scripts, this is probably wrong:

The shell this runs in doesn't have `set -e` nor any other reasonable way of error propagation, so when the configured file isn't readable by the `DynamicUser`, this code just prints e.g.

```
/nix/store/rwfwdk34fyz9qqyvn4vs0lnswscd73x0-plausible-setup: line 2: /var/lib/plausible/plausible-admin-user-password-file: Permission denied
/nix/store/rwfwdk34fyz9qqyvn4vs0lnswscd73x0-plausible-setup: line 3: /var/lib/plausible/plausible-secret-keybase-file: Permission denied
```

and then happily goes on with its day, with `ADMIN_USER_PWD` and `SECRET_KEY_BASE` set to the empty string.

Reply:
Even if it had `set -e`, it would still be wrong, in the same way as the scripts of Elixir itself here:

https://github.com/elixir-lang/elixir/issues/11114

Reply:
Fixes in PR https://github.com/NixOS/nixpkgs/pull/130062